extensions/amp-app-banner/1.0/component/android.js

```diff
@@ -0,0 +1,104 @@
+import {docInfoService} from '#preact/services/document';
+import {platformService} from '#preact/services/platform';
+import {timerService} from '#preact/services/timer';
+import {urlService} from '#preact/services/url';
+import {viewerService} from '#preact/services/viewer';
+import {xhrService} from '#preact/services/xhr';
+
+import {user} from '#utils/log';
+
+import {openWindowDialog} from '../../../../src/open-window-dialog';
+
+const OPEN_LINK_TIMEOUT = 1500;
+
+/**
+ * @return {{openOrInstall: function(): void, promise: Promise<Response>}|null}
+ */
+export function getAndroidAppInfo() {
+  const win = self.window;
+  // We want to fallback to browser builtin mechanism when possible.
+  const isChromeAndroid =
+    platformService.isAndroid() && platformService.isChrome();
+  const isProxyOrigin = urlService.isProxyOrigin(win.location);
+  const canShowBuiltinBanner =
+    !isProxyOrigin && !viewerService.isEmbedded() && isChromeAndroid;
+
+  if (canShowBuiltinBanner) {
+    user().info(
+      'BENTO-APP-BANNER',
+      'Browser supports builtin banners. Not rendering amp-app-banner.'
+    );
+    return null;
+  }
+
+  const manifestLink = win.document.head.querySelector(
+    'link[rel=manifest],link[rel=origin-manifest]'
+  );
```

Comment:
I tried, but get this (incredibly helpful!) error:
> ESLint: using scopedQuerySelector here is actually unnecessary, since you don't use child selector semantics.(local/query-selector)